src/services/relay_renewal_job/mod.rs

```diff
@@ -0,0 +1,73 @@
+use {
+    crate::{error::NotifyServerError, metrics::Metrics},
+    chrono::Duration,
+    relay_client::http::Client,
+    relay_rpc::{auth::ed25519_dalek::Keypair, domain::Topic},
+    sqlx::PgPool,
+    std::{future::Future, sync::Arc},
+    tokio::time,
+    tracing::{error, info, instrument},
+    url::Url,
+};
+
+mod refresh_topic_subscriptions;
+mod register_webhook;
+
+pub async fn start(
+    key_agreement_topic: Topic,
+    notify_url: Url,
+    keypair: Keypair,
+    relay_client: Arc<Client>,
+    postgres: PgPool,
+    metrics: Option<Metrics>,
+) -> Result<impl Future<Output = ()>, NotifyServerError> {
+    let period = Duration::days(1);
+
+    let mut interval = time::interval(period.to_std().expect("Should be able to convert to STD"));
+
+    // We must be able to run the job once on startup or we are non-functional
+    job(
+        key_agreement_topic.clone(),
+        &notify_url,
+        &keypair,
+        &relay_client,
+        &postgres,
+        metrics.as_ref(),
+    )
+    .await?;
+
+    Ok(async move {
+        tokio::time::sleep(interval.period()).await;
```

Comment:
Are you sure this is needed?

Reply:
Yeah it's needed since `tick()` doesn't delay the first time it's called, so doing a manual delay here.

I refactored this a little to make it more clear in 375efd973d37673ca8c6d660330fbe4de1372131